src/Microsoft.AspNetCore.Mvc.Core/ControllerBase.cs

```diff
@@ -1163,6 +1163,47 @@ public virtual FileContentResult File(byte[] fileContents, string contentType, s
         }
 
         /// <summary>
+        /// Returns a file with the specified range in the given <paramref name="fileContents" /> as content (<see cref="StatusCodes.Status206PartialContent"/> or
+        /// <see cref="StatusCodes.Status416RangeNotSatisfiable"/> if the range is not satisfiable), and the specified <paramref name="contentType" /> as the Content-Type.
+        /// </summary>
+        /// <param name="fileContents">The file contents.</param>
+        /// <param name="contentType">The Content-Type of the file.</param>
+        /// <param name="lastModified">The <see cref="DateTimeOffset"/> of when the file was last modified.</param>
+        /// <param name="entityTag">The <see cref="EntityTagHeaderValue"/> associated with the file.</param>
+        /// <returns>The created <see cref="FileContentResult"/> for the response.</returns>
+        [NonAction]
+        public virtual FileContentResult File(byte[] fileContents, string contentType, DateTimeOffset? lastModified, EntityTagHeaderValue entityTag)
+        {
+            return new FileContentResult(fileContents, contentType)
+            {
+                LastModified = lastModified,
+                EntityTag = entityTag,
+            };
+        }
+
+        /// <summary>
+        /// Returns a file with the specified range in the given <paramref name="fileContents" /> as content (<see cref="StatusCodes.Status206PartialContent"/> or
```

Comment:
If lastModified and etag are not specified, the `PreconditionState` is `Unspecified` and the range may still be served if the Range header is correctly provided. So that additional comment for 

> "This supports range requests." 

is relevant for all file controller actions in `ControllerBase`